Assets/Project/Scripts/MenuSelectScene/LevelSelect/RoadController.cs

```diff
@@ -49,21 +57,59 @@ private void Start()
             SetPointPosition();
         }
 
+        public void Reset()
+        {
+            PlayerPrefs.DeleteKey(PlayerPrefsKeys.ROAD + _roadId.ToString());
+        }
+
+        public void UpdateReleased()
+        {
+            _released = PlayerPrefs.GetInt(PlayerPrefsKeys.ROAD + _roadId.ToString(), Default.ROAD_RELEASED) == 1;
+
+            if (!_released) {
+                if (_constraintTrees.Length == 0) {
+                    // 初期状態で解放されている道
+                    _released = true;
+                } else {
+                    var constraintTreeClear = true;
+                    foreach (var constraintTree in _constraintTrees) {
+                        Debug.Log(constraintTree.GetComponent<TreeController>().cleared);
+                        constraintTreeClear = constraintTreeClear && constraintTree.GetComponent<TreeController>().cleared;
+                    }
+                    _released = constraintTreeClear;
```

Comment:
```suggestion
                    _released = _constraintTrees.All(tree => tree.GetComponent<TreeController>().cleared);
```
`using System.Linq;` を忘れずに

Reply:
最近Linqに慣れてきた。
助かります。